app/src/main/java/com/banuba/sdk/example/quickstart_c_api/MainActivity.kt

```diff
@@ -0,0 +1,214 @@
+package com.banuba.sdk.example.quickstart_c_api
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.opengl.GLSurfaceView
+import android.os.Bundle
+import android.os.Environment
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import com.banuba.sdk.example.common.BANUBA_CLIENT_TOKEN
+import com.banuba.sdk.utils.ContextProvider
+import kotlinx.android.synthetic.main.activity_main.*
+import java.io.*
+import java.nio.ByteBuffer
+import java.util.zip.ZipFile
+import javax.microedition.khronos.egl.EGLConfig
+import javax.microedition.khronos.opengles.GL10
+
+
+class MainActivity : AppCompatActivity() {
+    companion object {
+        private const val TAG = "MainActivity"
+    }
+
+    var banubaSdk = BanubaSdk()
+
+    lateinit var glView: GLSurfaceView
+    lateinit var render: EPRenderer
+
+    var onceInit = true
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        ContextProvider.setContext(applicationContext)
+        super.onCreate(savedInstanceState)
+
+        val pathToResources = application.filesDir.absolutePath + "/bnb-resources"
+        prepareResources(pathToResources)
+        prepareEffects(pathToResources)
+
+        banubaSdk.setOepListener{ image: ByteArray, width: Int, height: Int ->
+            runOnUiThread {
+                if (onceInit) {
+                    setContentView(R.layout.activity_main)
+                    onceInit = false
+                }
+
+                val bmp = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
+                bmp.copyPixelsFromBuffer(ByteBuffer.wrap(image))
+                // show processing result
+                imageView.setImageBitmap(bmp)
+            }
+        }
+
+        render = EPRenderer(banubaSdk, pathToResources, BANUBA_CLIENT_TOKEN)
+
+        glView = GLSurfaceView(this)
+        glView.setEGLContextClientVersion(3)
+        glView.setRenderer(render)
+        setContentView(glView)
+
+        val photo = application.assets.open("img/photo.jpg")
+        val bitmap = BitmapFactory.decodeStream(photo)
+
+        // process image async in RenderThread with GL context
+        render.processImage(bitmap)
+    }
+
+    override fun onDestroy() {
+        render.onDestroy()
+        super.onDestroy()
+    }
+
+    private fun prepareResources(resPath: String) = try {
+        val zipFile = File("$resPath.zip")
+
+        /* copy zip file from assets, if zip file is not exists */
+        if (!zipFile.exists()) {
+            val asset = application.assets.open("bnb-resources.zip")
+            val fout = zipFile.outputStream()
+            asset.copyTo(fout)
+            fout.close()
+            asset.close()
+        }
+
+        val destDir = File("$resPath")
+        /* exstract all files in zip to destDir, if destDir is not exists */
+        if (!destDir.exists()) {
+            destDir.mkdirs()
+            unzip(destDir, zipFile);
+        } else {
```

Comment:
Empty else?

Reply:
Yes. This is required for kotlin. Otherwise I get an error